cmds/sort/sort_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2016 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// TODO: test multi-file input
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+type Test struct {
+	flag string
+	in   string
+	out  string
+}
+
+var sortTests = []Test{
+	// already sorted, in == out
+	{"", "a\nb\nc\n", "a\nb\nc\n"},
+	// sort letters
+	{"", "c\na\nb\n", "a\nb\nc\n"},
+	// sort lexicographic
+	{"", "abc \nab\na bc\n", "a bc\nab\nabc \n"},
+	// sort without terminating newline
+	{"", "a\nb\nc", "a\nb\nc\n"},
+	// sort with utf-8 characters
+	{"", "γ\nα\nβ\n", "α\nβ\nγ\n"},
+	// reverse sort
+	{"-r", "c\na\nb\n", "c\nb\na\n"},
+	// reverse sort without terminating newline
+	{"-r", "a\nb\nc", "c\nb\na\n"},
+}
+
+func TestSort(t *testing.T) {
+	// Create temp directory
+	tmpDir, err := ioutil.TempDir("", "TestSort")
+	if err != nil {
+		t.Fatal("TempDir failed: ", err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	// Compile sort program
+	sortPath := filepath.Join(tmpDir, "sort")
+	out, err := exec.Command("go", "build", "-o", sortPath, "sort.go").CombinedOutput()
+	if err != nil {
+		t.Fatalf("go build -o %v cmds/sort: %v\n%s", sortPath, err, string(out))
+	}
+
+	// Table-driven testing
+	for _, test := range sortTests {
+		// Two ways to test:
+		pipes(t, test, tmpDir, sortPath)
+		files(t, test, tmpDir, sortPath)
+	}
+}
+
+func pipes(t *testing.T, test Test, tmpDir string, sortPath string) {
+	cmd := fmt.Sprintf("printf %#v | %s %s", test.in, sortPath, test.flag)
```

Comment:
Would it be better to do perform the pipe in go itself rather than relying on sh command being there?